src/ansys/fluent/core/solver/flobject.py

```diff
@@ -388,6 +399,43 @@ class Real(SettingsBase[RealType], Numerical):
     expression values.
     """
 
+    def get_state_as_quantity(self) -> Optional[ansys_units.Quantity]:
+        """Get the state of the object as a Quantity."""
+        error = None
+        if not ansys_units:
+            error = "Code not configured to support units."
+        if not error:
+            try:
```

Comment:
Probably want to tighten the scope of this try down to the Quantity instantiation.